coordinator/points_writer.go

```diff
@@ -380,15 +381,18 @@ func (w *PointsWriter) writeToShard(shard *meta.ShardInfo, database, retentionPo
 	if err == tsdb.ErrShardNotFound {
 		err = w.TSDBStore.CreateShard(database, retentionPolicy, shard.ID, true)
 		if err != nil {
-			w.Logger.Info(fmt.Sprintf("write failed for shard %d: %v", shard.ID, err))
-
+			if w.Diagnostic != nil {
```

Comment:
Not a huge fan of this nil check on the diagnostic, but I guess we have to :/

Reply:
The primary reason for having these nil checks everywhere is to avoid making a nil version of the interface for every package and attempting to use it as a default. The old code didn't need it because I set the logger to the null logger. I'm removing that instead and using this pattern.

Reply:
I see. Maybe have a Diagnostic thats local to this package that does something like this

```go
type LocalDiag struct {
Diag Diagnostic
}

func (d *LocalDiag) DiagMethod(...) {
if d == nil {
return
}

d.Diag.DiagMethod(...)
}
```

So that we can avoid having these nil checks everywhere. I don't feel super strongly about it, but having 

```
if d.diag != nil {
}
```
everywhere definitely is an eye sore